pkg/console/controllers/plugins/controller.go

```diff
@@ -0,0 +1,189 @@
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorv1.Managed:
+		klog.V(4).Infof("console-operator is in a managed state: migrating available plugins")
+	case operatorv1.Unmanaged:
+		klog.V(4).Infof("console-operator is in an unmanaged state: skipping available plugins migration")
+		return nil
+	case operatorv1.Removed:
+		klog.V(4).Infof("console-operator is in a removed state: removing migrated plugins")
+		return c.removeMigratedPlugins(ctx, operatorConfig)
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	availablePluginsName, err := c.GetAvailablePluginsName()
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("AvailablePluginMigration", "FailedAvailablePluginsGet", err))
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+	migratedPluginsArray, err := GetMigratedPlugins(operatorConfig)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("AvailablePluginMigration", "FailedMigratedPluginsGet", err))
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	pluginsToEnable := GetPluginsToEnable(operatorConfig, availablePluginsName, migratedPluginsArray)
+
+	if len(pluginsToEnable) == 0 {
+		return statusHandler.FlushAndReturn(nil)
+	}
+
+	err = c.UpdateOperatorConfig(ctx, operatorConfig, pluginsToEnable, migratedPluginsArray)
+	statusHandler.AddConditions(status.HandleProgressingOrDegraded("AvailablePluginMigration", "FailedOperatorConfigUpdate", err))
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	return statusHandler.FlushAndReturn(err)
+}
+
+func (c *PluginsMigrationController) GetAvailablePluginsName() ([]string, error) {
+	availablePlugins, err := c.consolePluginLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	var availablePluginsName []string
+	for _, plugin := range availablePlugins {
+		availablePluginsName = append(availablePluginsName, plugin.Name)
+	}
+	return availablePluginsName, nil
+}
+
+func GetMigratedPlugins(operatorConfig *operatorv1.Console) ([]string, error) {
+	migratedPluginsAnnotation, migratedPluginsAnnotationExists := operatorConfig.Annotations[MigratedPluginsAnnotation]
+	migratedPluginsArray := []string{}
+	if migratedPluginsAnnotationExists {
+		err := json.Unmarshal([]byte(migratedPluginsAnnotation), &migratedPluginsArray)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return migratedPluginsArray, nil
+}
+
+func GetPluginsToEnable(
+	operatorConfig *operatorv1.Console,
+	availablePluginsName []string,
+	migratedPluginsArray []string,
+) []string {
+	var pluginsToEnable []string
+	for _, availablePluginName := range availablePluginsName {
+		if !util.ContainsString(migratedPluginsArray, availablePluginName) && !util.ContainsString(operatorConfig.Spec.Plugins, availablePluginName) {
+			pluginsToEnable = append(pluginsToEnable, availablePluginName)
```

Comment:
We don't want to enable all plugins. We only want to enable ones from a well-defined allow list (e.g., `["kubevirt"]`)